java/code/src/com/suse/manager/webui/controllers/VirtualGuestsController.java

```diff
@@ -331,15 +348,54 @@ public static ModelAndView edit(Request request, Response response, User user) {
         data.put("inSSM", RhnSetDecl.SYSTEMS.get(user).contains(hostId));
 
         /* For the rest of the template */
+        MinionController.addActionChains(user, data);
         data.put("guestUuid", guestUuid);
         data.put("isSalt", host.hasEntitlement(EntitlementManager.SALT));
         return new ModelAndView(data, "virtualization/guests/edit.jade");
     }
 
+    /**
+     * Display the New Virtual Guest page
+     *
+     * @param request the request
+     * @param response the response
+     * @param user the user
+     * @return the ModelAndView object to render the page
+     */
+    public static ModelAndView create(Request request, Response response, User user) {
+        Map<String, Object> data = new HashMap<>();
+
+        Long hostId = Long.parseLong(request.params("sid"));
+        Server host = SystemManager.lookupByIdAndUser(hostId, user);
+
+        if (!host.hasEntitlement(EntitlementManager.SALT)) {
+            Spark.halt(HttpStatus.SC_BAD_REQUEST, "Only for Salt-managed virtual hosts");
+        }
+
+        /* For system-common.jade */
+        data.put("server", host);
+        data.put("inSSM", RhnSetDecl.SYSTEMS.get(user).contains(hostId));
+
+        /* For the rest of the template */
+        MinionController.addActionChains(user, data);
+        data.put("isSalt", host.hasEntitlement(EntitlementManager.SALT));
+
+        return new ModelAndView(data, "virtualization/guests/create.jade");
```

Comment:
Due to this [PR](https://github.com/uyuni-project/uyuni/pull/506), after rebasing with master you will need to adapt this path with `templates/virtualization/guests/create.jade`